src/FfmpegOutputNode.cpp

```diff
@@ -0,0 +1,99 @@
+#include "FfmpegOutputNode.h"
+#include <QDebug>
+#include <QJsonObject>
+#include <QGuiApplication>
+
+FfmpegOutputNode::FfmpegOutputNode(Context *context, QSize chainSize)
+    : OutputNode(context, chainSize)
+    , m_recording(false)
+    , m_ffmpegArguments({"-vcodec", "h264", "output.mp4"})
+{
+
+}
+
+FfmpegOutputNode::FfmpegOutputNode(const FfmpegOutputNode &other)
+    : OutputNode(other) {
+}
+
+FfmpegOutputNode::~FfmpegOutputNode() {
+    setRecording(false);
+}
+
+void FfmpegOutputNode::setRecording(bool recording) {
+    if (m_recording == recording)
+        return;
+
+    m_recording = recording;
+    if (m_recording) {
+        QSize size = chain()->size();
+        QString sizeStr(QString("%1x%2").arg(
+                    QString::number(size.width()),
+                    QString::number(size.height())));
+
+        m_pixelBuffer.resize(3 * size.width() * size.height());
+
+        m_ffmpeg.start("ffmpeg", QStringList() 
+                << "-y"
+                << "-vcodec" << "rawvideo"
+                << "-f" << "rawvideo"
+                << "-pix_fmt" << "rgb24"
+                << "-s" << sizeStr
+                << "-i" << "pipe:0"
+                << "-vf" << "vflip"
+                << m_ffmpegArguments);
+    } else {
+        m_ffmpeg.closeWriteChannel();
+        m_ffmpeg.waitForFinished();
+        //qInfo() << m_ffmpeg.readAllStandardOutput();
+        //qInfo() << m_ffmpeg.readAllStandardError();
+    }
+    emit recordingChanged(m_recording);
+}
+
+bool FfmpegOutputNode::recording() {
+    return m_recording;
+}
+
+QStringList FfmpegOutputNode::ffmpegArguments() {
+    return m_ffmpegArguments;
+}
+
+void FfmpegOutputNode::setFfmpegArguments(QStringList ffmpegArguments) {
+    m_ffmpegArguments = ffmpegArguments;
+    emit ffmpegArgumentsChanged(m_ffmpegArguments);
+}
+
+QString FfmpegOutputNode::typeName() {
+    return "FfmpegOutputNode";
+}
+
+VideoNode *FfmpegOutputNode::deserialize(Context *context, QJsonObject obj) {
+    // TODO: You should be able to change the size of an OutputNode after
+    // it has been created. For now this is hard-coded
+    FfmpegOutputNode *e = new FfmpegOutputNode(context, QSize(128, 128));
+    return e;
+}
+
+bool FfmpegOutputNode::canCreateFromFile(QString filename) {
+    return false;
+}
+
+VideoNode *FfmpegOutputNode::fromFile(Context *context, QString filename) {
+    return nullptr;
+}
+
+QMap<QString, QString> FfmpegOutputNode::customInstantiators() {
+    auto m = QMap<QString, QString>();
+    m.insert("FfmpegOutput", "FfmpegOutputInstantiator.qml");
+    return m;
+}
+
+void FfmpegOutputNode::recordFrame() {
```

Comment:
TODO: Actually record without manually calling `recordFrame`

Reply:
Should it just have a `QTimer` to call this periodically?

Reply:
Yeah, and probably a dedicated thread...